docs/guide/topics/permissions.rst

```diff
@@ -0,0 +1,219 @@
+
+    # If you wanted to create a new admin role, that has the `Administrator` permission, you can do it like so:
+    permissions = discord.Permissions(administrator=True)
+
+    # However, the underlying value for this permission (or `bit` as the API docs explain it) can be viewed by:
+    permissions.value
+    >>> 8
+
+    # You can also instantiate a Permissions class with an integer that represents the bit(s) you want to provide, like so:
+    permissions = discord.Permissions(268435634)
+    # This Permissions instance has the `view_audit_log`, `manage_server`, `manage_roles`, `manage_channels` and `kick_members` permissions.
+
+
+Using Permissions to create a new role
+---------------------------------------
+
+So we want to make a role for a new game in our guild, that we can add to users to ping them for news and such. Let's do that now:
+
+.. code-block:: python3
+
+    async def create_new_role(name: str, mentionable: bool, guild: discord.Guild) -> discord.Role:
+        await guild.create_role(name=name, mentionable=mentionable, reason="New game!")
+
+This rather short example just creates a new role within the :class:`~Guild` passed to this method.
+
+Let's flesh it out more, and use the :meth:`~Guild.create_role` method to it's full capability with a set of :class:`Permissions`.
+
+.. code-block:: python3
+
+    # So we want a new role, but let's give it permissions to do some things.
+    # We will define a set of `Permissions` that do what we need:
+    permissions = discord.Permissions()
+    permissions.send_messages = True
+    permissions.add_reactions = True
+    permissions.create_public_threads = True
+
+    # Above is now a working Permissions instance with the name `permissions`, we could also define it like so:
+    permissions = discord.Permissions(send_messages=True, add_reactions=True, create_public_threads=True)
+
+    # and we pass this to the role creation method!
+    await guild.create_role(name="My cool game", mentionable=True, permissions=permissions, reason="New game!")
+
+Now we have a brand new role, named ``"My cool game"`` and it now have the permissions we outlined above.
+
+Setting permissions on an existing channel
+-------------------------------------------
+
+Let's say you wanted to edit the permissions of an existing channel, perhaps to give a role more access.
+An example here is that my new moderator role needs the Manage Messages permission in the #general channel.
+We will be using the :meth:`~TextChannel.set_permissions` method:
+
+.. code-block:: python3
+
+    # Let's create a PermissionOverwrite instance that has some explicit permissions set:
+    overwrites = discord.PermissionOverwrite()
+    overwrites.update(send_messages=True, send_messages_in_threads=True)
+
+    # and now we update that with `manage_messages`:
+    overwrites.manage_messages = True
+
+    # we can now set those permissions on the channel:
+    await channel.set_permissions(moderator_role, overwrites=overwrites)
```

Comment:
[TextChannel.set_permissions](https://discordpy.readthedocs.io/en/master/api.html#discord.TextChannel.set_permissions) takes `overwrite`, not `overwrites`
```suggestion
    await channel.set_permissions(moderator_role, overwrite=overwrites)
```
The name of the var itself should probably also be adjusted.